### ISSUE TITLE
Inconsistent path matching when two resources share a prefix

### ISSUE THREAD

**Issue by kekbur** [quarkusio/quarkus#30854]
### Describe the bug

When there are two resource classes with a shared path prefix, RESTEasy Reactive incorrectly returns a 404 response. This behavior is different in RESTEasy classic, where requests to both resources return a 200 response.

```java
@Path("/chat")
public class CollectionResource {
    @GET
    @Path("/messages")
    public String messages() {
        return "Get multiple messages";
    }
}

@Path("/chat/messages/{messageId}")
public class SingleResource {
    @GET
    public String message(@PathParam("messageId") String messageId) {
        return "Get one message: " + messageId;
    }
}
```

### Expected behavior

I'm not sure what the correct behavior is according to JAX-RS spec. In my opinion requests to `/chat/messages` and `/chat/messages/1` should return a 200 response with both RESTEasy Classic and RESTEasy Reactive.

### Actual behavior

 Request | JAX-RS implementation | Response |
| --- | --- | --- |
| GET /chat/messages | RESTEasy Reactive | 404 |
| GET /chat/messages/1 | RESTEasy Reactive | 200 |
| GET /chat/messages | RESTEasy Classic | 200 |
| GET /chat/messages/1 | RESTEasy Classic | 200 |

### How to Reproduce?

Minimal reproducing project: [code-with-quarkus.zip](https://github.com/quarkusio/quarkus/files/10580546/code-with-quarkus.zip)

Test failure:
`mvn -f pom.xml clean test`

Tests succeed:
`mvn -f pom-resteasy-classic.xml clean test`

### Output of `uname -a` or `ver`

CYGWIN_NT-10.0-22000 DESKTOP-5J3QD5D 3.4.3-1.x86_64 2022-12-16 12:38 UTC x86_64 Cygwin

### Output of `java -version`

java version "17.0.3.1" 2022-04-22 LTS Java(TM) SE Runtime Environment (build 17.0.3.1+2-LTS-6) Java HotSpot(TM) 64-Bit Server VM (build 17.0.3.1+2-LTS-6, mixed mode, sharing)

### Quarkus version or git rev

2.16.1.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.6

### Additional information

As an additional data point, the resource paths on the Quarkus 404 error page and in the dev UI.

![image](https://user-images.githubusercontent.com/10086020/216651377-02e5a283-b3b8-4f2d-ad9b-b85cc448f2e5.png)
![image](https://user-images.githubusercontent.com/10086020/216651511-81e1f795-7675-4a34-8a52-b98f63449df0.png)

**Comment by gefloh** [comment]
There are already some of these like e.g. #27154 or #30667 and it might already have been fixed with #30695.

**Comment by geoand** [comment]
@gefloh thanks for the detailed update on the current state of affairs!

This issue is actually a duplicate of https://github.com/quarkusio/quarkus/issues/26496 which has **not** been addressed yet.
